crates/bevy_asset/src/server/info.rs

```diff
@@ -111,7 +111,10 @@ impl AssetInfos {
         .unwrap()
     }
 
-    #[allow(clippy::too_many_arguments)]
+    #[expect(
+        clippy::too_many_arguments,
+        reason = "Arguments needed so that both `create_loading_handle_untyped()` and `get_or_create_path_handle_internal()` may share code."
+    )]
     fn create_handle_internal(
```

Comment:
An issue should be created to reduce the complexity of this function.

Reply:
pub(crate) fn create_loading_handle_untyped(
        &mut self,
        type_id: TypeId,
        type_name: &'static str,
    ) -> UntypedHandle {
        unwrap_with_context(
            (|| -> Result<UntypedHandle, GetOrCreateHandleInternalError> {
                let provider = &self
                    .handle_providers
                    .get(&type_id)
                    .ok_or(MissingHandleProviderError(type_id))?;
                Ok(Self::create_handle_internal(
                    provider,
                    &mut self.infos,
                    &mut self.living_labeled_assets,
                    self.watching_for_changes,
                    None,
                    None,
                    true,
                ))
            })(),
            type_name,
        )
        .unwrap()
    }

    fn create_handle_internal(
        provider: &&AssetHandleProvider,
        infos: &mut HashMap<UntypedAssetId, AssetInfo>,
        living_labeled_assets: &mut HashMap<AssetPath<'static>, HashSet<Box<str>>>,
        watching_for_changes: bool,
        path: Option<AssetPath<'static>>,
        meta_transform: Option<MetaTransform>,
        loading: bool,
    ) -> UntypedHandle {
        if watching_for_changes {
            if let Some(path) = &path {
                let mut without_label = path.to_owned();
                if let Some(label) = without_label.take_label() {
                    let labels = living_labeled_assets.entry(without_label).or_default();
                    labels.insert(label.as_ref().into());
                }
            }
        }
        let handle = provider.reserve_handle_internal(true, path.clone(), meta_transform);
        let mut info = AssetInfo::new(Arc::downgrade(&handle), path);
        if loading {
            info.load_state = LoadState::Loading;
            info.dep_load_state = DependencyLoadState::Loading;
            info.rec_dep_load_state = RecursiveDependencyLoadState::Loading;
        }
        infos.insert(handle.id, info);
        UntypedHandle::Strong(handle)
    }

...

            Entry::Vacant(entry) => {
                let should_load = match loading_mode {
                    HandleLoadingMode::NotLoading => false,
                    HandleLoadingMode::Request | HandleLoadingMode::Force => true,
                };
                let provider = &self
                    .handle_providers
                    .get(&type_id)
                    .ok_or(MissingHandleProviderError(type_id))?;
                let handle = Self::create_handle_internal(
                    provider,
                    &mut self.infos,
                    &mut self.living_labeled_assets,
                    self.watching_for_changes,
                    Some(path),
                    meta_transform,
                    should_load,
                );
                entry.insert(handle.id());
                Ok((handle, should_load))
            }

Reply:
The lint doesn't trigger with this quick change

Reply:
Looks like Alice has you covered! #15322